src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -225,9 +225,11 @@ func (q *qemuAmd64) enableProtection() error {
 		q.qemuMachine.Options += "kvm-type=tdx,confidential-guest-support=tdx"
 		q.kernelParams = append(q.kernelParams, Param{"tdx_guest", ""})
 		virtLog.WithFields(logrus.Fields{
-			"subsystem":     "qemuAmd64",
-			"machine":       q.qemuMachine,
-			"kernel-params": q.kernelParameters}).
+			"subsystem":               "qemuAmd64",
+			"machine":                 q.qemuMachine,
+			"kernel-params-debug":     q.kernelParamsDebug,
+			"kernel-params-non-debug": q.kernelParamsNonDebug,
+			"kernel-params":           q.kernelParams}).
```

Comment:
How about changing this slightly to remove the duplication. Something like:

```diff
diff --git a/src/runtime/virtcontainers/qemu_amd64.go b/src/runtime/virtcontainers/qemu_amd64.go
index 4dbddd732..5eaadb8db 100644
--- a/src/runtime/virtcontainers/qemu_amd64.go
+++ b/src/runtime/virtcontainers/qemu_amd64.go
@@ -217,6 +217,13 @@ func (q *qemuAmd64) enableProtection() error {
 		return err
 	}
 
+	logger := virtLog.WithFields(logrus.Fields{
+		"subsystem":               "qemuAmd64",
+		"machine":                 q.qemuMachine,
+		"kernel-params-debug":     q.kernelParamsDebug,
+		"kernel-params-non-debug": q.kernelParamsNonDebug,
+		"kernel-params":           q.kernelParams})
+
 	switch q.protection {
 	case tdxProtection:
 		if q.qemuMachine.Options != "" {
@@ -224,26 +231,14 @@ func (q *qemuAmd64) enableProtection() error {
 		}
 		q.qemuMachine.Options += "kvm-type=tdx,confidential-guest-support=tdx"
 		q.kernelParams = append(q.kernelParams, Param{"tdx_guest", ""})
-		virtLog.WithFields(logrus.Fields{
-			"subsystem":               "qemuAmd64",
-			"machine":                 q.qemuMachine,
-			"kernel-params-debug":     q.kernelParamsDebug,
-			"kernel-params-non-debug": q.kernelParamsNonDebug,
-			"kernel-params":           q.kernelParams}).
-			Info("Enabling TDX guest protection")
+		logger.Info("Enabling TDX guest protection")
 		return nil
 	case sevProtection:
 		if q.qemuMachine.Options != "" {
 			q.qemuMachine.Options += ","
 		}
 		q.qemuMachine.Options += "confidential-guest-support=sev"
-		virtLog.WithFields(logrus.Fields{
-			"subsystem":               "qemuAmd64",
-			"machine":                 q.qemuMachine,
-			"kernel-params-debug":     q.kernelParamsDebug,
-			"kernel-params-non-debug": q.kernelParamsNonDebug,
-			"kernel-params":           q.kernelParams}).
-			Info("Enabling SEV guest protection")
+		logger.Info("Enabling SEV guest protection")
 		return nil
 
 	// TODO: Add support for other x86_64 technologies
```

Reply:
@jodh-intel I have updated as your suggestion, thank you.